web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -475,6 +477,16 @@ public Object start(Element config, ServiceContext context) throws Exception {
             Integer dbHeartBeatFixedDelay = Integer.parseInt(handlerConfig.getValue(Geonet.Config.DB_HEARTBEAT_FIXEDDELAYSECONDS, "60"));
             createDBHeartBeat(context.getResourceManager(), gnContext, dbHeartBeatInitialDelay, dbHeartBeatFixedDelay);
         }
+
+        boolean linkMonitorEnabled = Boolean.parseBoolean(handlerConfig.getValue(Geonet.Config.LINK_MONITOR_ENABLED, "false"));
```

Comment:
Extract method `configureLinkMonitor()`?  

Probably applies to DB heartbeat above too.  Not sure what our stand is on refactoring upstream code is, but IMO, a class with over 1000 lines is *probably* just a *little* too big :fearful: